src/main.ts

```diff
@@ -13,24 +13,62 @@ export const URI = 'Dataway';
 
 export type URI = typeof URI;
 
+/**
+ * Interface usualy used as default value for dataway,
+ * contain no data
+ */
 export interface NotAsked {
   readonly _tag: 'NotAsked';
 }
 
+/**
+ * Interface used to represent a remote datasource being loaded or
+ * a command is being processed (ex: form submit)
+ */
 export interface Loading {
   readonly _tag: 'Loading';
 }
-
+/**
+ * Interface used to represent any failure regarding request of
+ * a remote datasource
+ * can be used to
+ * 1. represent and store an http error
+ * 2. represent and store a value decoding/validation error
+ * 3. a computation error on the retrieved data
+ * @typeparam E the expected error type
+ */
 export interface Failure<E> {
   readonly _tag: 'Failure';
   readonly failure: E;
 }
 
+/**
+ * Interface used to represent remote datasource fetching success
+ * Store the retrieved value.
+ * It can be used to
+ * 1. reprensent and store the value that was successfuly retrieved
+ * 2. represent and store successfull transformation applied to the retrieved value
+ * @typeparam A the expected value type
+ */
 export interface Success<A> {
   readonly _tag: 'Success';
   readonly success: A;
 }
 
+/**
+ * This type represent the four possible state of a remote datasource
+ * fetching can result.
```

Comment:
Dans la foulée de la phrase de Nico, ça nous donne : 
`This type represents the four possible states of a remote datasource fetching result`